### PR TITLE
Changed workflow of TradeQueue processing into sagas + code cleanup

### PR DESCRIPTION
**Pull request by polonzer**
https://lykkex.atlassian.net/browse/LWDEV-4628

### CODE REVIEW

File: src/Lykke.Job.TransactionHandler/Projections/FeeProjection.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            _log = log ?? throw new ArgumentNullException(nameof(log));
+            _feeLogRepository = feeLogRepository ?? throw new ArgumentNullException(nameof(feeLogRepository));
+        }
+
+        // todo: remove?
+        public async Task Handle(TradeCreatedEvent evt)
+        {
+            await _log.WriteInfoAsync(nameof(FeeProjection), nameof(TradeCreatedEvent), evt.ToJson(), "");
+
+            ChaosKitty.Meow();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/FeeProjection.cs]: Meow before CreateAsync is useless

File: src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs [hunk 2]
```diff
@@ -17,73 +18,172 @@ namespace Lykke.Job.TransactionHandler.Projections
...
+            _ethereumTransactionRequestRepository = ethereumTransactionRequestRepository ?? throw new ArgumentNullException(nameof(ethereumTransactionRequestRepository));
         }
 
-        public async Task Handle(IssueTransactionStateSavedEvent evt)
+        public async Task Handle(TransferOperationStateSavedEvent evt)
         {
-            await _log.WriteInfoAsync(nameof(OperationHistoryProjection), nameof(IssueTransactionStateSavedEvent), evt.ToJson(), "");
+            await _log.WriteInfoAsync(nameof(OperationHistoryProjection), nameof(TransferOperationStateSavedEvent), evt.ToJson(), "");
+
+            ChaosKitty.Meow();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs]: Meow before state changing is useless

File: src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs [hunk 3]
```diff
@@ -17,73 +18,172 @@ namespace Lykke.Job.TransactionHandler.Projections
...
+                    Amount = amount,
+                    TransactionId = transactionId,
+                    IsHidden = false,
+                    AddressFrom = destWallet?.Address,
+                    AddressTo = destWallet?.MultiSig,
+                    Multisig = destWallet?.MultiSig,
+                    IsSettled = false,
+                    State = transferState
+                });
+
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs]: Meow will be good here

File: src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs [hunk 4]
```diff
@@ -17,73 +18,172 @@ namespace Lykke.Job.TransactionHandler.Projections
...
+                    Amount = -amount,
+                    TransactionId = transactionId,
+                    IsHidden = false,
+                    AddressFrom = sourceWallet?.Address,
+                    AddressTo = sourceWallet?.MultiSig,
+                    Multisig = sourceWallet?.MultiSig,
+                    IsSettled = false,
+                    State = transferState
+                });
+        }
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs]: ...and here

File: src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs [hunk 5]
```diff
@@ -17,73 +18,172 @@ namespace Lykke.Job.TransactionHandler.Projections
...
+                    nameof(RegisterOperation), operation.ToJson(),
+                    $"Unexpected response from Operations Service: {response.ToJson()}");
+            }
+        }
+
+        public async Task Handle(TradeCreatedEvent evt)
+        {
+            await _log.WriteInfoAsync(nameof(OperationHistoryProjection), nameof(TradeCreatedEvent), evt.ToJson(), "");
+
+            ChaosKitty.Meow();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs]: Meow before state changing again

File: src/Lykke.Job.TransactionHandler/Projections/OrdersProjection.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            _log = log ?? throw new ArgumentNullException(nameof(log));
+            _marketOrdersRepository = marketOrdersRepository ?? throw new ArgumentNullException(nameof(marketOrdersRepository));
+        }
+
+        public async Task Handle(TradeCreatedEvent evt)
+        {
+            await _log.WriteInfoAsync(nameof(OrdersProjection), nameof(TradeCreatedEvent), evt.ToJson(), "");
+
+            ChaosKitty.Meow();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/OrdersProjection.cs]: ...and again

File: src/Lykke.Job.TransactionHandler/Projections/OrdersProjection.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+            try
+            {
+                await _marketOrdersRepository.CreateAsync(evt.MarketOrder);
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/OrdersProjection.cs]: switch to the TryInsert inside?

File: src/Lykke.Job.TransactionHandler/Projections/TransfersProjection.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            _log = log ?? throw new ArgumentNullException(nameof(log));
+            _transferLogRepository = transferLogRepository ?? throw new ArgumentNullException(nameof(transferLogRepository));
+        }
+
+        public async Task Handle(TransferOperationStateSavedEvent evt)
+        {
+            await _log.WriteInfoAsync(nameof(TransfersProjection), nameof(TransferOperationStateSavedEvent), evt.ToJson(), "");
+
+            ChaosKitty.Meow();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Projections/TransfersProjection.cs]: one more meow out of place

File: src/Lykke.Job.TransactionHandler/Sagas/Services/ContextFactory.cs [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                var aggregatedTrades = await AggregateSwaps(trades);
+
+                context.SellTransfer = aggregatedTrades.sellTransfer;
+                context.BuyTransfer = aggregatedTrades.buyTransfer;
+
+                var operations = new[] { aggregatedTrades.sellTransfer, aggregatedTrades.buyTransfer };
+
+                foreach (var operation in operations)
+                {
+                    var trade = context.ClientTrades.FirstOrDefault(x =>
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Sagas/Services/ContextFactory.cs]: Is field ClientTrades contains only active client trades? This linear search looks not scalable

File: src/Lykke.Job.TransactionHandler/Sagas/Services/ContextFactory.cs [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+                        x.ClientId == clientId && x.AssetId == operation.Asset.Id &&
+                        Math.Abs(x.Amount - (double)operation.Amount) < 0.00000001);
+
+                    // find existed operation (which was inserted in LW after guarantee transfer)
+                    var existed = context.Operations.FirstOrDefault(x => x.ClientId == clientId && x.AssetId == operation.Asset.Id);
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.TransactionHandler/Sagas/Services/ContextFactory.cs]: And this too

File: src/Lykke.Job.TransactionHandler/Handlers/OperationsCommandHandler.cs [hunk 11]
```diff
@@ -5,77 +5,129 @@
...
             _assetsServiceWithCache = assetsServiceWithCache ?? throw new ArgumentNullException(nameof(assetsServiceWithCache));
             _walletCredentialsRepository = walletCredentialsRepository ?? throw new ArgumentNullException(nameof(walletCredentialsRepository));
-            _bitcoinTransactionsRepository = bitcoinTransactionsRepository ?? throw new ArgumentNullException(nameof(bitcoinTransactionsRepository));
         }
 
-        public async Task<CommandHandlingResult> Handle(Commands.RegisterCashInOutOperationCommand command)
+        public async Task<CommandHandlingResult> Handle(Commands.SaveTransferOperationStateCommand operationStateCommand, IEventPublisher eventPublisher)
         {
-            await _log.WriteInfoAsync(nameof(OperationsCommandHandler), nameof(Commands.RegisterCashInOutOperationCommand), command.ToJson(), "");
+            await _log.WriteInfoAsync(nameof(OperationsCommandHandler), nameof(Commands.SaveTransferOperationStateCommand), operationStateCommand.ToJson());
```
Comment (mmusin) [inline src/Lykke.Job.TransactionHandler/Handlers/OperationsCommandHandler.cs]: _log.WriteInfo() ?

File: src/Lykke.Job.TransactionHandler/Sagas/Services/ContextFactory.cs (hunk 9 above)
Comment (polonzer) [reply src/Lykke.Job.TransactionHandler/Sagas/Services/ContextFactory.cs]: This collection contains just 2 elements.

File: src/Lykke.Job.TransactionHandler/Sagas/LimitTradeSaga.cs [hunk 12]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Lykke.Job.TransactionHandler.Sagas
```
Comment (paradise) [inline src/Lykke.Job.TransactionHandler/Sagas/LimitTradeSaga.cs]: is this really needed?

**Pull request merged**